homework/index.js

```diff
@@ -30,17 +32,94 @@
     return elem;
   }
 
+  const root = document.getElementById('root');
+  const header = createAndAppend('header', root, { class: 'header' });
+  createAndAppend('p', header, { text: 'HYF Repositories' });
+
+  const select = createAndAppend('select', header, { id: 'selector' });
+  const option = x => {
+    for (let i = 0; i < x.length; i++) {
+      createAndAppend('option', select, { text: x[i].name, value: i });
+    }
+  };
+
+  const div = createAndAppend('div', root, { id: 'container' });
+  const leftContainer = createAndAppend('div', div, { class: 'left-block frame' });
+  const table = createAndAppend('table', leftContainer);
+  const listContributor = createAndAppend('tbody', table);
+
+  const rightContainer = createAndAppend('div', div, { class: 'right-block frame' });
+  createAndAppend('p', rightContainer, {
+    text: 'Contributions',
+    class: 'contributor',
+  });
+
+  const list = createAndAppend('ul', rightContainer, { class: 'list1' });
+
+  function renderContributors(selRep) {
+    list.innerHTML = '';
+
+    for (let i = 0; i < selRep.length; i++) {
+      const li = createAndAppend('li', list, { class: 'contributor-item' });
+      const linkFor = createAndAppend('a', li, { target: '_blank', href: selRep[i].html_url });
+      createAndAppend('img', linkFor, { src: selRep[i].avatar_url, class: 'avatar', height: 48 });
+      const divCont = createAndAppend('div', linkFor, { class: 'contributor-data' });
+      createAndAppend('div', divCont, { text: selRep[i].login });
+      createAndAppend('div', divCont, {
+        text: selRep[i].contributions,
+        class: 'badge',
+      });
+    }
+  }
+
+  function renderRepo(listContributors, repo) {
+    const content = [
+      { title: 'Repository', attribute: 'name' },
+      { title: 'Description', attribute: 'description' },
+      { title: 'Forks', attribute: 'forks' },
+      { title: 'Updated', attribute: 'updated_at' },
+    ];
+    listContributors.innerHTML = '';
+
+    for (let i = 0; i < content.length; i++) {
+      const headTitle = createAndAppend('tr', listContributors);
+      createAndAppend('td', headTitle, { text: `${content[i].title} :`, class: 'label' });
+      const cellContent = createAndAppend('td', headTitle);
+      if (content[i].attribute === 'name') {
+        createAndAppend('a', cellContent, {
+          href: repo.html_url,
+          text: repo.name,
+          target: '_blank',
+        });
+      } else {
+        cellContent.textContent = repo[content[i].attribute];
+      }
+    }
+    fetchJSON(repo.contributors_url, (error, contributors) => renderContributors(contributors));
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
+      const root1 = document.getElementById('root');
```

Comment:
Why did you have to rename the variable here as `root1` instead of `root`?

Reply:
It did not work that time I thought that was the reason, But I also changed some other things may be that changed effect it.
Let me update it end see what will happen.

Reply:
instead of root1 just I rote as root .that is what you mean?

Reply:
Yes. That's correct. You can rename it back to `root`.